### PR TITLE
Added Set timeout, required to increase the timeout value

### PR DESCRIPTION
**Pull request by mimmi20**
This PR replaces PR #171. It includes all changes from there and the fixed review comments.

### CODE REVIEW

**Comment by mimmi20** [comment]
@asgrim Could you check and merge this?

File: src/BrowscapUpdater.php [hunk 1]
```diff
@@ -154,6 +154,20 @@ public function getLogger()
     }
 
     /**
+     * Sets the Connection Timeout
+     *
+     * @param int $connectTimeout
+     *
+     * @return \BrowscapPHP\BrowscapUpdater
```
Comment (asgrim) [inline src/BrowscapUpdater.php]: Probably more future-proof to annotate this as `@return self`

File: src/BrowscapUpdater.php [hunk 2]
```diff
@@ -154,6 +154,20 @@ public function getLogger()
...
+     */
+    public function setConnectTimeout($connectTimeout)
```
Comment (asgrim) [inline src/BrowscapUpdater.php]: No unit tests for this method

File: src/BrowscapUpdater.php [hunk 3]
```diff
@@ -154,6 +154,20 @@ public function getLogger()
...
+    {
+        $this->connectTimeout = (int) $connectTimeout;
+
+        return $this;
```
Comment (asgrim) [inline src/BrowscapUpdater.php]: Is it absolutely necessary to return `$this`? See: https://ocramius.github.io/blog/fluent-interfaces-are-evil/

Comment (mimmi20) [inline src/BrowscapUpdater.php]: fixed

File: src/BrowscapUpdater.php (hunk 2 above)
Comment (mimmi20) [inline src/BrowscapUpdater.php]: fixed

File: src/BrowscapUpdater.php (hunk 1 above)
Comment (mimmi20) [inline src/BrowscapUpdater.php]: `@return` removed

File: src/BrowscapUpdater.php [hunk 4]
```diff
@@ -121,22 +120,16 @@ public function setCache($cache)
                 Exception::CACHE_INCOMPATIBLE
             );
         }
-
-        return $this;
```
Comment (asgrim) [inline src/BrowscapUpdater.php]: Ah, this breaks BC unnecessarily, please revert; I only meant to not use fluent on new features

File: tests/BrowscapUpdaterTest.php [hunk 5]
```diff
@@ -76,7 +76,7 @@ public function testSetGetCache()
             ->disableOriginalConstructor()
             ->getMock();
 
-        self::assertSame($this->object, $this->object->setCache($cache));
+        $this->object->setCache($cache);
```
Comment (asgrim) [inline tests/BrowscapUpdaterTest.php]: Test changes need to be reverted too to ensure we don't break BC accidentally in future :)

Comment (mimmi20) [inline tests/BrowscapUpdaterTest.php]: fixed

**Pull request merged**